packages/nimble-components/src/wafer-map/modules/zoom-handler.ts

```diff
@@ -6,33 +6,41 @@ import {
     ZoomTransform,
     zoomTransform
 } from 'd3-zoom';
-import type { DataManager } from './data-manager';
-import type { RenderingModule } from './rendering';
+import type { Dimensions, ZoomEvent, ZoomHandlerData } from '../types';
 
 /**
  * ZoomHandler deals with user interactions and events like zooming
  */
 export class ZoomHandler {
+    public onZoom: ((event: ZoomEvent) => void) | undefined;
     private zoomTransform: ZoomTransform = zoomIdentity;
     private readonly minScale = 1.1;
     private readonly minExtentPoint: [number, number] = [-100, -100];
     private readonly extentPadding = 100;
-    private zoomBehavior: ZoomBehavior<Element, unknown> | undefined;
+    private readonly zoomBehavior: ZoomBehavior<Element, unknown>;
+    private readonly canvas: HTMLCanvasElement;
+    private readonly zoomContainer: HTMLElement;
+    private readonly containerDimensions: Dimensions | undefined;
+    private readonly canvasLength: number;
+    private readonly renderingFunction: VoidFunction;
```

Comment:
Why is `renderingFunction` a variable instead of a method? It doesn't look like it's passed around anywhere by reference. Making it a method would be easier to follow and allow removing the `this.renderingFunction === undefined` check below.

Reply:
I agree with removing the check - it was indeed some leftover code left from a previous iteration of the code (good catch here). It is no longer necessary. However, the `renderingFunction` is a property since it is expected to to populated by the corresponding method from the `renderModule` ... Of course, this could be wrapped in a function, but I did not see yet a strong case for it....